lib/pengine/unpack.c

```diff
@@ -4908,11 +4909,15 @@ add_node_attrs(const xmlNode *xml_obj, pcmk_node_t *node, bool overwrite,
                      CRM_ATTR_UNAME, node->priv->name);
 
     pcmk__insert_dup(node->priv->attrs, CRM_ATTR_ID, node->priv->id);
-    if (pcmk__str_eq(node->priv->id, scheduler->dc_uuid, pcmk__str_casei)) {
+
+    if ((scheduler->dc_node == NULL)
```

Comment:
These changes should be fine as long as the DC can't change without a new scheduler object being created.

And it should be impossible in practice for `CRM_ATTR_IS_DC` not to get set to either true or false, as long as nothing else ever manipulates `scheduler->dc_node`, clears `node->priv->attrs`, or alters the `CRM_ATTR_IS_DC` value...

Reply:
Yep the DC is parsed from the scheduler input early and doesn't change. The input can't change during the lifetime of a scheduler object (after being expanded early on).